src/ripple/app/tx/impl/XChainBridge.cpp

```diff
@@ -0,0 +1,1627 @@
+}
+
+NotTEC
+XChainCommit::preflight(PreflightContext const& ctx)
+{
+    if (!ctx.rules.enabled(featureXChainBridge))
+        return temDISABLED;
+
+    if (auto const ret = preflight1(ctx); !isTesSuccess(ret))
+        return ret;
+
+    if (ctx.tx.getFlags() & tfUniversalMask)
+        return temINVALID_FLAG;
+
+    auto const amount = ctx.tx[sfAmount];
+
+    if (amount.signum() <= 0 || !isLegalNet(amount))
+        return temBAD_AMOUNT;
+
+    return preflight2(ctx);
+}
+
+TER
+XChainCommit::preclaim(PreclaimContext const& ctx)
+{
+    auto const bridge = ctx.tx[sfXChainBridge];
+    auto const amount = ctx.tx[sfAmount];
+
+    auto const sleB = ctx.view.read(keylet::bridge(bridge));
+    if (!sleB)
+    {
+        return tecNO_ENTRY;
+    }
+
+    AccountID const thisDoor = (*sleB)[sfAccount];
+    AccountID const account = ctx.tx[sfAccount];
+
+    if (thisDoor == account)
+    {
+        // Door account can't lock funds onto itself
+        return tecXCHAIN_SELF_COMMIT;
+    }
+
+    bool isLockingChain = false;
+    {
+        if (thisDoor == bridge.lockingChainDoor())
+            isLockingChain = true;
+        else if (thisDoor == bridge.issuingChainDoor())
+            isLockingChain = false;
+        else
+            return tecINTERNAL;
+    }
+
+    if (isLockingChain)
+    {
+        if (bridge.lockingChainIssue() != ctx.tx[sfAmount].issue())
+            return tecBAD_XCHAIN_TRANSFER_ISSUE;
+    }
+    else
+    {
+        if (bridge.issuingChainIssue() != ctx.tx[sfAmount].issue())
+            return tecBAD_XCHAIN_TRANSFER_ISSUE;
+    }
+
+    return tesSUCCESS;
+}
+
+TER
+XChainCommit::doApply()
+{
+    PaymentSandbox psb(&ctx_.view());
+
+    auto const account = ctx_.tx[sfAccount];
+    auto const amount = ctx_.tx[sfAmount];
+    auto const bridge = ctx_.tx[sfXChainBridge];
+
+    auto const sle = psb.peek(keylet::account(account));
+    if (!sle)
+        return tecINTERNAL;
+
+    auto const sleB = psb.read(keylet::bridge(bridge));
+    if (!sleB)
+        return tecINTERNAL;
+
+    auto const dst = (*sleB)[sfAccount];
+
+    auto const thTer = transferHelper(
+        psb,
+        account,
+        dst,
+        /*dstTag*/ std::nullopt,
+        /*claimOwner*/ std::nullopt,
+        amount,
+        TransferHelperCanCreateDst::no,
+        ctx_.journal);
+
+    if (!isTesSuccess(thTer))
+        return thTer;
+
+    psb.apply(ctx_.rawView());
+
+    return tesSUCCESS;
+}
+
+//------------------------------------------------------------------------------
+
+NotTEC
+XChainCreateClaimID::preflight(PreflightContext const& ctx)
+{
+    if (!ctx.rules.enabled(featureXChainBridge))
+        return temDISABLED;
+
+    if (auto const ret = preflight1(ctx); !isTesSuccess(ret))
+        return ret;
+
+    if (ctx.tx.getFlags() & tfUniversalMask)
+        return temINVALID_FLAG;
+
+    auto const reward = ctx.tx[sfSignatureReward];
+
+    if (!isXRP(reward) || reward.signum() < 0 || !isLegalNet(reward))
```

Comment:
On [line 439](https://github.com/XRPLF/rippled/pull/4292/files#diff-f59c1290a2b9bb42469fda9b064384dbe2e9f7376f7937bf3fab225241c15fafR439) the signature amount must be strictly positive, but here it must be only nonnegative. Shouldn't those be aligned?

Reply:
Also, couldn't someone run a bridge with a signature reward amount of `0`?

Reply:
Fixed in 91de824b2 [FOLD] Address PR feedback 1: (allowed for zero reward)